src/Microsoft.OpenApi.Readers/ParsingContext.cs

```diff
@@ -45,18 +45,21 @@ internal OpenApiDocument Parse(YamlDocument yamlDocument, OpenApiDiagnostic diag
             {
                 VersionService = new OpenApiV2VersionService();
                 doc = this.VersionService.LoadDocument(this.RootNode);
+                diagnostic.SpecificationVersion = OpenApiSpecVersion.OpenApi2_0;
             }
             else if (inputVersion.StartsWith("3.0."))
             {
                 this.VersionService = new OpenApiV3VersionService();
                 doc = this.VersionService.LoadDocument(this.RootNode);
+                diagnostic.SpecificationVersion = OpenApiSpecVersion.OpenApi3_0;
             }
             else
```

Comment:
The `else if (...) { }` and `else { ...}` has the same codes? why cannot we merge them together? 

@PerthCharern @darrelmiller

Reply:
The "else" is for a default case. It is, but doesn't have to be, the same as the "else if". This makes sure we list out the cases for 2.0 and 3.0 explicitly.

If you are worried about code duplication, we can refactor the guts out.

---
In reply to: [160842325](https://github.com/Microsoft/OpenAPI.NET/pull/175#discussion_r160842325) [](ancestors = 160842325)

Reply:
Won't fix with the scope of this change.

---
In reply to: [160843959](https://github.com/Microsoft/OpenAPI.NET/pull/175#discussion_r160843959) [](ancestors = 160843959,160842325)